Sources/_CryptoExtras/RSA/RSA_boring.swift

```diff
@@ -315,6 +345,95 @@ extension BoringSSLRSAPublicKey {
             return output
         }
 
+        fileprivate func blind<H: HashFunction>(
+            _ message: _RSA.BlindSigning.PreparedMessage,
+            parameters: _RSA.BlindSigning.Parameters<H>
+        ) throws -> _RSA.BlindSigning.BlindingResult {
+            let rsaPublicKey = CCryptoBoringSSL_EVP_PKEY_get0_RSA(self.pointer)
+            let modulusByteCount = Int(CCryptoBoringSSL_RSA_size(rsaPublicKey))
+            let e = try ArbitraryPrecisionInteger(copying: CCryptoBoringSSL_RSA_get0_e(rsaPublicKey))
+            let n = try ArbitraryPrecisionInteger(copying: CCryptoBoringSSL_RSA_get0_n(rsaPublicKey))
+            let finiteField = try FiniteFieldArithmeticContext(fieldSize: n)
+
+            // 1. encoded_msg = EMSA-PSS-ENCODE(msg, bit_len(n)) with Hash, MGF, and salt_len as defined in the parameters
+            // 2. If EMSA-PSS-ENCODE raises an error, re-raise the error and stop
+            // 3. m = bytes_to_int(encoded_msg)
+            let m = try BlindSigningHelpers.EMSAPSSEncode(
+                rsaPublicKey: rsaPublicKey,
+                modulusByteCount: modulusByteCount,
+                message: message,
+                parameters: parameters
+            )
+
+            // 4. c = is_coprime(m, n)
+            let c = try m.isCoprime(with: n)
+
+            // 5. If c is false, raise an "invalid input" error and stop
+            if !c { throw CryptoKitError(_RSA.BlindSigning.ProtocolError.invalidInput) }
+
+            // 6. r = random_integer_uniform(1, n)
+            // 7. inv = inverse_mod(r, n)
+            // 8. If inverse_mod fails, raise a "blinding error" error and stop
+            // NOTE: We retry here until we get an appropriate r, which is suggested.
+            let (r, inv) = try {
+                while true {
+                    let r = try ArbitraryPrecisionInteger.random(inclusiveMin: 1, exclusiveMax: n)
+                    guard let inv = try finiteField.inverse(r) else { continue }
+                    return (r, inv)
+                }
+            }()
```

Comment:
OK, I've updated it to use `var`, although I had this construction to ensure a `let` binding because during the last round of review, in addition to dealing with pointers, one of the comments was keeping track of which state had been computed and which was mutable.